uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -803,15 +803,20 @@ private static function validateAbsenceTypeIsActiveAndValid($params, $absenceTyp
    *
    * @param int $contactID
    * @param \CRM_HRLeaveAndAbsences_BAO_PublicHoliday $publicHoliday
+   * @param boolean $nonDeletedOnly
+   *   When true will only return result if the given public
+   *   holiday request is not yet deleted.
    *
    * @return \CRM_HRLeaveAndAbsences_BAO_LeaveRequest|null
    */
-  public static function findPublicHolidayLeaveRequest($contactID, PublicHoliday $publicHoliday) {
+  public static function findPublicHolidayLeaveRequest($contactID, PublicHoliday $publicHoliday, $nonDeletedOnly = TRUE) {
```

Comment:
We should avoid adding new params like this as much as possible. The reason is that now the method can do two different things. 
I know that, sometimes, adding a new param is a way to avoid having to duplicate a huge query, but here the method is so simple that it's ok to have a separate method to get only non-deleted public holiday leave requests

Reply:
That means we would have two identical methods with only a line being the difference between both methods? The parameter seems to me as a filter that filters off some records and we have many of such methods in L&A.

Reply:
Yeah, that's exactly what I meant. We do have methods like that in L&A and they're bad as well. Unfortunately, sometimes these methods are so complex that duplicating them would be risky, so adding a new param sounds better. However, in this case here, the code is so simple that the duplication is fine compared with the readability improvement we'll have. 

Just for reference, here are some places where we duplicated code to improved readability:
The [AbsencePeriod::getCurrentPeriod()](https://github.com/civicrm/civihr/blob/99ed22a5709a4d11ed6c8b7bb2739c19b4d08890/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php#L300-L310) method is exactly the same as the [AbsencePeriod::getPeriodOverlappingDate()](https://github.com/civicrm/civihr/blob/99ed22a5709a4d11ed6c8b7bb2739c19b4d08890/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php#L320-L332)

The methods [AbsenceType::getEnableAbsenceTypes()](https://github.com/civicrm/civihr/blob/99ed22a5709a4d11ed6c8b7bb2739c19b4d08890/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php#L608-L620) and [AbsenceType::getEnabledSicknessAbsenceTypes()](https://github.com/civicrm/civihr/blob/99ed22a5709a4d11ed6c8b7bb2739c19b4d08890/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php#L627-L634) are another interesting example. Technically, we don't really need the `getEnabledSickenessAbsenceTypes()` methods. We could have just added a new param to `getEnabledAbsenceTypes()` and that would work the same way, but the readbility is much better with the second method.

I'm sure that if you look at the code there are some more examples of this